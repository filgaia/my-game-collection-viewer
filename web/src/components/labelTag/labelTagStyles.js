const labelTagStyles = (theme) => ({
    chip: {
        marginLeft: '10px',
        '&:hover': {
            fontWeight: 'bold',
            boxShadow: '2px 2px 1px 0px rgba(0,0,0,0.5)'
        }
    },
    drawerChip: {
        marginLeft: 0,
        width: '100%'
    },
    seletedChip: {
        fontWeight: 'bold'
    }
});

export default labelTagStyles;
