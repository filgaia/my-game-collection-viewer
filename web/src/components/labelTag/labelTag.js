// @vendors
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
// @material
import {
    Chip,
    withStyles
} from '@material-ui/core';
// @styles
import labelTagStyles from './labelTagStyles';
// @utilities
import { tagCodeToColor } from '../../utilities/index';
// @constants
import { CARD_OPACITY } from './../../constants/index';

class LabelTag extends Component {

    constructor(props) {
        super(props);

        this.handleClickFilter = this.handleClickFilter.bind(this);
    }

    handleClickFilter() {
        const { gamesInformation, setLabelFilter, label } = this.props;
        if (gamesInformation.get('idLabelFilter') === label.id) {
            setLabelFilter(null);
        } else {
            setLabelFilter(label.id);
        }
    }

    render() {
        const { classes, isDrawer, label, gamesInformation } = this.props;

        const labelSeleted = gamesInformation.get('idLabelFilter') === label.id;
        const opacity = labelSeleted ? 1 : CARD_OPACITY;
        const tagColor = tagCodeToColor(label.background_color, opacity);

        const classTag = classNames(
            classes.chip,
            {
                [classes.seletedChip]: labelSeleted,
                [classes.drawerChip]: isDrawer
            }
        );

        return (
            <Chip className={classTag}
                style={{ backgroundColor: tagColor }}
                onClick={this.handleClickFilter}
                key={`label-${label.id}`}
                label={label.name}
                clickable
            />
        );
    }
}

LabelTag.propTypes = {
    classes: PropTypes.object.isRequired,
    gamesInformation: PropTypes.object.isRequired,
    isDrawer: PropTypes.bool,
    label: PropTypes.object.isRequired,
    setLabelFilter: PropTypes.func.isRequired
};

export default withStyles(labelTagStyles)(LabelTag);
