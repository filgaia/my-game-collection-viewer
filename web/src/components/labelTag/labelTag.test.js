import React from 'react';
import renderer from 'react-test-renderer';
import { LabelTag } from './labelTag';

describe('<LabelTag />', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<LabelTag
        classes={{}}
        gamesInformation={{ get: jest.fn() }}
        label={{}}
        setLabelFilter={jest.fn()}
      />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
