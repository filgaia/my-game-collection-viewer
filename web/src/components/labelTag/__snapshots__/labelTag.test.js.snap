// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LabelTag /> renders correctly 1`] = `
<div
  className="MuiButtonBase-root MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-clickable MuiChip-clickableColorDefault MuiChip-filledDefault css-1cjdymu-MuiButtonBase-root-MuiChip-root"
  onBlur={[Function]}
  onClick={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  role="button"
  style={
    Object {
      "backgroundColor": "rgba(0,0,0,1)",
    }
  }
  tabIndex={0}
>
  <span
    className="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
  />
</div>
`;
